toc.md

```diff
@@ -1,54 +1,76 @@
+* **ease of use**: a BJP should be simple to use out-of-the-box and integrate with rails in the context of building it in Ruby
+
 
 ## Introducing Workerholic: Overall Architecture
+
 ![Workerholic Overvall Architecture](/images/workerholic_overall_architecture.png)
 
-Above is a diagram of the overall architecture of our take on a background job processor. On the left is a web application that includes our library, the jobs are defined in there, the jobs are serialized and pushed into Redis into "Job Queues". On the right, Workerholic workers poll from the job queues to see if there are any jobs that need to be done; if there is, then the workers will use the "Job Processor" to do the jobs. Regardless of whether the job is completed successfully or not, we store the job back into Redis as "Stats" that we will show on our web UI (not shown here). If a job did fail, we use "Job Retry" together with the "Job Scheduler" to attempt to retry a job sometime in the future. A future timestamp is placed on the job, it gets pushed into Redis into a sorted set as "Scheduled Jobs", the Job Scheduler will peek the sorted set and compare timestamps to see if there is a job due. If there is then the job will be enqueued into a Job Queue and the cycle continues.
+Above is a diagram of the overall architecture of our BJP, Workerholic.
+- On the left is a stack of instances of a web application, each including Workerholic. Jobs are defined and enqueued in these web application instances.
+- In order for the jobs to be enqueued, they are first serialized and then stored in Redis into Lists, serving as `Job Queues`.
+- On the right, Workerholic workers poll from the job queues and, if there are any jobs to be done, process the jobs using the `Job Processor` component of Workerholic.
+- Regardless of whether the job is completed successfully or not, we store the job back in Redis inside a data structure serving as `Stats` storage that we will then use to show on our web UI.
+- If a job failed, we use the `Job Retry` component along with the `Job Scheduler` component in order to attempt to retry a job sometime in the future. To do so, a future timestamp is placed on the job, it is then stored into Redis into a `Scheduled Jobs` sorted set (a Redis data structure that we will expand on later in this post).
+- The `Job Scheduler` will peek the sorted set and compare timestamps to see if there is a job due. If there is then the job will be enqueued into a Job Queue and the cycle continues.
 
 ## Building Workerholic
-Let's start diving into the numerous features we wanted in Workerholic!
+
+The core of this project was to build a BJP from scratch and sharing our findings, what we learned and the challenges we faced. Next, we will dive into each feature that we deemed belonged into a BJP and how they are incorporated in Workerholic.
```

Comment:
The core (the core? maybe something like "the idea behind this project"?) of this project was to build a BJP from scratch and share our findings, what we learned and the challenges we faced. Next, we will dive into each feature that we deemed belonged into a BJP and how they are incorporated (maybe "implemented"? It just sounds better, IMO) in Workerholic.